addons/godot-next/inspector_plugins/behavior_collection_inspector_plugin.gd

```diff
@@ -0,0 +1,91 @@
+tool
+extends EditorInspectorPlugin
+
+##### PROPERTIES #####
+
+var class_type: ClassType
+
+##### NOTIFICATIONS #####
+
+func _init():
+	class_type = ClassType.new()
+
+##### VIRTUALS #####
+
+func can_handle(object):
+	return object is BehaviorCollection
+
+func parse_property(object, type, path, hint, hint_text, usage):
+	match type:
+		TYPE_ARRAY:
+			if hint == PROPERTY_HINT_RESOURCE_TYPE and hint_text.begins_with("#"):
+				add_custom_control(_generate_gui(object, path, hint_text))
+				return true
+		TYPE_DICTIONARY:
+			if hint == PROPERTY_HINT_RESOURCE_TYPE and hint_text.begins_with("#"):
+				add_custom_control(_generate_gui(object, path, hint_text))
+				return true
+
+##### PRIVATE METHODS #####
+
+func _find_inheritors(type_path) -> Array:
+	class_type.path = type_path
+	var list = class_type.get_deep_inheritors_list()
+	var type_map = class_type.get_deep_type_map()
+	var scripts = []
+	for a_name in list:
+		scripts.append(load(type_map[a_name].path))
+	return scripts
+
+func _generate_gui(object: Object, path: String, hint_text: String) -> Control:
+	
+	var hbox := HBoxContainer.new()
+	
+	var type_path := hint_text.replace("#", "")
+	var elements = _find_inheritors(type_path)
+	
+	var dropdown := _generate_type_dropdown(elements)
+	var button := _generate_add_button()
+	
+	button.connect("pressed", self, "_on_add_button_pressed", [object, path, dropdown])
+	
+	dropdown.size_flags_horizontal = HBoxContainer.SIZE_EXPAND_FILL
+	
+	hbox.add_child(dropdown)
+	hbox.add_child(button)
+	
+	return hbox
+
+func _generate_type_dropdown(elements: Array) -> Control:
+	var dropdown := OptionButton.new()
+	var i := 0
+	for script in elements:
+		class_type.path = script.resource_path
+		dropdown.add_item(class_type.get_script_class(), i)
+		dropdown.set_item_metadata(i, script)
+		i += 1
+	
+	return dropdown
+
+func _generate_add_button() -> Control:
+	var button = ToolButton.new()
+	button.text = "+"
```

Comment:
Would prefer that this be an icon, but as we've discussed previously, I'll just go ahead an make this change after a merge.